contrib/caldav/calcurse-caldav.py

```diff
@@ -112,15 +181,25 @@ def remote_query(conn, cmd, path, additional_headers, body):
     if isinstance(body, str):
         body = body.encode('utf-8')
 
-    conn.request(cmd, path, headers=headers, body=body)
+    if authmethod == 'basic':
+        conn.request(cmd, path, headers=headers, body=body)
+        resp = conn.getresponse()
 
-    resp = conn.getresponse()
+        if not resp:
+            return (None, None)
 
-    if not resp:
-        return (None, None)
+        headers = resp.getheaders()
+        body = resp.read().decode('utf-8')
 
-    headers = resp.getheaders()
-    body = resp.read().decode('utf-8')
+    elif authmethod == 'oauth2':
+        # httplib2 can separate headers and body through simultaneous assignment
+        # Using headers and resp variables for compatibility
+        headers, body = conn.request(path, cmd, body=body, headers=headers)
+        resp = headers
+        body = body.decode('utf-8')
```

Comment:
Out of curiousity: Can't we just use httplib2 for both basic and OAuth2 authentication. Then we could drop some of this magic. If so, it might be a good idea to switch basic authentication to httplib2 in a preparatory ptach first, then add OAuth2 support.

Reply:
It definitely appears to have all the needed functionality that *http.client* has, including disabling SSL certificate validation for basic authentication. While it would definitely help clean things up, it comes with the downside of adding a dependency to basic authentication usage. It comes down to whether the refactor is worth adding that dependency.